### PR TITLE
fix(command-deploy): update alias description

### PR DESCRIPTION
**Pull request by rstavchansky**
**Summary**

Addresses docs-related feedback from a couple of different Support Forums posts:
- https://answers.netlify.com/t/using-alias-is-deploying-to-production-without-prod/33975
- https://answers.netlify.com/t/headers-not-read-for-alias-deploy-from-netlify-cli/40264

Addresses docs-specific concerns in https://github.com/netlify/cli/issues/1984

**Test plan**
View [the docs in the branch](https://github.com/netlify/cli/blob/docs/1984-update-alias-description/docs/commands/deploy.md).

**Note** This PR addresses docs feedback, but the [referenced issue](https://github.com/netlify/cli/issues/1984) contains suggestions for CLI warnings as well.

### CODE REVIEW

File: docs/commands/deploy.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ netlify deploy
 - `functions` (*string*) - Specify a functions folder to deploy
 - `prod` (*boolean*) - Deploy to production
 - `prodIfUnlocked` (*boolean*) - Deploy to production if unlocked, create a draft otherwise
-- `alias` (*string*) - Specifies the alias for deployment. Useful for creating predictable deployment URLs. Maximum 37 characters.
+- `alias` (*string*) - Specifies the alias for deployment, the string at the beginning of the deploy subdomain. Useful for creating predictable deployment URLs. Avoid setting an alias string to the same value as a deployed branch. `alias` doesn’t create a branch deploy and can’t be used in conjunction with the branch subdomain feature. Maximum 37 characters.
```
Comment (rstavchansky) [inline docs/commands/deploy.md:91]: Reviewer note: this is a lot longer than our normal description, but I'm attempting to help users avoid misconceptions about usage for this flag.

**Comment by rstavchansky** [comment]
I noted that I need to build the docs, but I ran `npm run docs` and got some errors including`Error: Cannot find module 'markdown-magic'.

**Comment by erezrokah** [comment]
> I noted that I need to build the docs, but I ran `npm run docs` and got some errors including`Error: Cannot find module 'markdown-magic'.

The docs are generated from another sub project, you'll need to install its' dependencies via `npm run site:build:install`

**Review by erezrokah — APPROVED** [review]
🚀

**Review by overlordofmu — COMMENTED** [review]
Looks good to me. I was asked to review but this looks like it was already merged. If it was not I would certainly have approved the change.

**Comment by rstavchansky** [comment]
Should have waited to get your 👍🏻 before merging @overlordofmu; thanks for taking a look.

**Pull request merged**